### ISSUE TITLE
Customer cannot set ticket priority

### ISSUE THREAD

**Issue by hindsa-nabeel** [SythilTech/Odoo#34]
Hi Steve,
I wonder if it is possible to allow the certain customer categories set his ticket priority, as in case of Business Functions Down or Server Down (Very High) etc... It would be great if a checkbox to add or remove this functionality is available at the setup/configuration level.

Of course the Support agent may reset the priority sighting an SLA or other reason, etc. 

Best regards.

**Comment by SythilTech** [comment]
Hello,

The ability for the customer to assign ticket priority was introduced in version 1.2.13 for Odoo 11, it added a selection so you can choose if priority is available to public or only registered users and should display on the website form.

Unfortunately this functionality is not on a per category bases though so it is not possible to make it so the priority field only displays for certain categories.

Hopefully the existing functionality is sufficient for your requirements

Regards
Steven

**Comment by hindsa-nabeel** [comment]
Thanks Steven,
I was hoping if some sort of profiling can be introduced. Alas, may be it is not meant for now. I hope you will be able to address this requirement some time in the future.

p.s. I found the password reset checkbox, but could not find the priority setting in v1.3.2!!

It would be great if this page is updated such that it reflects the current release:
[https://www.odoo.com/apps/modules/11.0/website_support/](url)

Best regards.

**Comment by hindsa-nabeel** [comment]
Hi Steven,
I uninstalled Website_Support 1,3,2 and reinstalled 1,2.13. It is possible to set the priority as a customer but through the CRM module.

I then wanted to upgrade to 1.3.2 and I got an error, then decided to upgrade to 1.2.14, but still got this error:

Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/odoo11/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo11/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/opt/odoo/odoo11/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/opt/odoo/odoo11/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo11/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo11/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo11/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo11/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo11/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/odoo11/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/opt/odoo/odoo11/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo11/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo11/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-48>", line 2, in button_immediate_upgrade
  File "/opt/odoo/odoo11/odoo/addons/base/module/module.py", line 71, in check_and_log
    return method(self, *args, **kwargs)
  File "/opt/odoo/odoo11/odoo/addons/base/module/module.py", line 603, in button_immediate_upgrade
    return self._button_immediate_function(type(self).button_upgrade)
  File "/opt/odoo/odoo11/odoo/addons/base/module/module.py", line 542, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/opt/odoo/odoo11/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/odoo11/odoo/modules/loading.py", line 376, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check)
  File "/opt/odoo/odoo11/odoo/modules/loading.py", line 274, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/opt/odoo/odoo11/odoo/modules/loading.py", line 135, in load_module_graph
    migrations.migrate_module(package, 'pre')
  File "/opt/odoo/odoo11/odoo/modules/migration.py", line 178, in migrate_module
    migrate(self.cr, installed_version)
  File "/opt/odoo/odoo11/custom-addons/website_support/migrations/11.0.1.2.14/pre-migrate.py", line 2, in migrate
    cr.execute('ALTER TABLE website_support_ticket RENAME COLUMN ticket_number TO ticket_number_old')
  File "/opt/odoo/odoo11/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/odoo/odoo11/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: column "ticket_number_old" of relation "website_support_ticket" already exists

In the end.

If I am to install 1.3.2 afresh, I do not seem to find where to configure the availability of the priority field to the customer. 

Your help is much appreciated.

Best regards.

**Comment by SythilTech** [comment]
Hello,

You can allow customers to set priority under the Customer Support->Configuration->Settings menu with a user with Support Manager group access in version 1.3.2.

In version 1.2.13 the setting would be under CRM configuration accessible to Sales Managers

Hmm the error is an interesting one as version 1.2.14 introduction a migration script, I'm guessing this is caused by the downgrade then upgrade cycle.

You might want to temporary remove the migrations folder to circumvent this issue while re-upgrading the module.

Also the documentation was been updated in version 1.3.0 to reflect the menu changes introduced in version 1.2.14 (previous version)
https://www.odoo.com/apps/modules/11.0/website_support/

**Comment by hindsa-nabeel** [comment]
Thanks Steven,
I managed to set the priority field through the Customer Support menu after the creation of the support manager, but it took me forever... 

Thank you for updating the Help instructions on the website, but I wonder if there is a manual advising on how to benefit from the various functions your development offers.

You are right, the error happened after a upgrade and downgrade then upgrade again. It is now fixed and back onto 1.3.2.

This issue is now closed. 

Best regards and keep up the good work.

Nabeel.
